### PR TITLE
dv smokes

### PR DESCRIPTION
**Pull request by mmajerni**
//test: `@dv`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  VirtualizationList.java:5: Wrong order for 'io.syndesis.qe.fragments.common.list.RowList' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  VirtualizationList.java:8: 'org.openqa.selenium.By' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  VirtualizationList.java:10: Wrong order for 'com.codeborne.selenide.Condition.exist' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:4: 'io.cucumber.datatable.DataTable' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:5: Wrong order for 'io.syndesis.qe.pages.SyndesisPageObject' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:7: 'lombok.extern.slf4j.Slf4j' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:8: Wrong order for 'org.openqa.selenium.By' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:13: Wrong order for 'com.codeborne.selenide.Condition.matchText' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/pages/Virtualizations/
  Virtualizations.java:16: Wrong order for 'org.assertj.core.api.Assertions.fail' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:6: 'cucumber.api.java.en.Then' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:9: Wrong order for 'io.syndesis.qe.pages.Virtualizations.Virtualizations' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:12: 'lombok.extern.slf4j.Slf4j' should be separated from previous imports. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:13: Wrong order for 'org.openqa.selenium.By' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:19: Wrong order for 'com.codeborne.selenide.Selenide.$' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:21: Wrong order for 'org.assertj.core.api.Assertions.assertThat' import. [ImportOrder]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:92: 'lambda arguments' has incorrect indentation level 16, expected level should be 12. [Indentation]
ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/
  DataVirtualizationSteps.java:93: 'lambda arguments' has incorrect indentation level 16, expected level should be 12. [Indentation]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:1078:22: '=' is not followed by whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:1078:23: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:1078:24: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mkralik3** [comment]
Once https://github.com/syndesisio/syndesis-qe/pull/1326 will be merged, please, rebase with master and update enabling/disabling DV addon in the DV resource class. 

Deploy step is already created here: https://github.com/syndesisio/syndesis-qe/blob/master/utilities/src/main/java/io/syndesis/qe/bdd/CommonSteps.java#L123 . It needs to update only dv resource class ( https://github.com/syndesisio/syndesis-qe/blob/master/utilities/src/main/java/io/syndesis/qe/resource/impl/DV.java ) and define how to properly deploy / undeploy DV. Undeploying happens automatically after the test. An example for PublicApi: https://github.com/syndesisio/syndesis-qe/blob/391da0d631737cca2093774213fc9c176ed50aef/utilities/src/main/java/io/syndesis/qe/resource/impl/PublicOauthProxy.java

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:74:8: Unused import - java.util.concurrent.TimeUnit. [UnusedImports]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Connection - CRUD | Create and delete 
Connection - CRUD | Kebab menu 
Connection - CRUD | Kebab menu edit and view 
Integration - DB to DB | Read and update operations 
Integration info | Check error for integration with the same name 
Integration info | Check error for integration which has same name after update 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    @When("create an invalid view and check that error appears")
+    public void createAnInvalidViewAndCheckThatErrorAppears() {
+        final String SAVE_BUTTON = "*[data-testid=\"ddl-editor-save-button\"]";
+        SelenideElement editor = $(By.className("ddl-editor"));
+        editor.$(By.className("CodeMirror")).click();
+        editor.$("textarea").sendKeys("Lorem Ipsum");
+        editor.$(By.cssSelector(SAVE_BUTTON)).click();
+
+        try {
+            Thread.sleep(1000);
```
Comment (mkralik3) [inline ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java]: Use `TestUtils.sleepIgnoreInterrupt` instead

File: ui-tests/src/test/resources/features/other/data-virtualization.feature [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    When edit data virtualization "smokeVirtualization"
+    And edit data virtualization view "testView"
+
+    Then check that ddl exists and contains text "CREATE VIEW testView"
+    And check that number of rows for preview is 1 at view editor page
+    When create an invalid view and check that error appears
+
+    And click on the "Done" button
+
+    Then make action "Stop" on the virtualization "smokeVirtualization"
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/other/data-virtualization.feature]: When, Then is using for validation step

File: ui-tests/src/test/resources/features/other/data-virtualization.feature [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    When set parameters for SQL client
+      | todo | 10 | 2 |
+    And click on the "Submit" button
+    Then check that number of rows for query is 0 in sql client
+
+    When navigate to the "Data" page
+    And make action "Publish" on the virtualization "smokeVirtualization"
+
+    Then wait until virtualization "smokeVirtualization" gets into "Running" state
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/other/data-virtualization.feature]: When, Then is using for validation step

File: ui-tests/src/test/resources/features/other/data-virtualization.feature [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    And click on the "Done" button
+
+    Then check that data virtualization view "contact" is present in view list
+    And check that data virtualization view "todo" is present in view list
+
+    When go to "SQL client" tab on virtualization page
+    And set parameters for SQL client
+      | contact | 10 | 2 |
+    And click on the "Submit" button
+    Then check that number of rows for query is 0 in sql client
```
Comment (mkralik3) [inline ui-tests/src/test/resources/features/other/data-virtualization.feature]: I would suggest checking whether the SQL client shows any data without offset and after that test whether offset works. Also, you should not rely on data which is in the `contact` table by default ` Joe | Jackson` . I suggest to add this into the background phase:
```
    And reset content of "contact" table
    And inserts into "contact" table
      | Joe | Jackson | Red Hat | db |
```

File: ui-tests/src/test/resources/features/other/data-virtualization.feature [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# @sustainer: mmajerni@redhat.com
+@dv
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply ui-tests/src/test/resources/features/other/data-virtualization.feature]: added

File: ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    /*
+     * State != starting state
+     * State is state of integration. e.g. "Running", "Stopped"
+     */
+    @Then("^wait until virtualization \"([^\"]*)\" gets into \"([^\"]*)\" state$")
+    public void waitForIntegrationState(String virtualizationName, String virtualizationStatus) {
+        commonSteps.navigateTo("Data");
+        SelenideElement virtualization = virtualizations.getVirtualization(virtualizationName);
+        TestUtils.sleepForJenkinsDelayIfHigher(10);
+        assertThat(TestUtils.waitForEvent(
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java]: It is created based on IntegrationSteps: 
https://github.com/syndesisio/syndesis-qe/blob/0795500d970fdf5fde78dc5810cadef4179e67e0/ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java#L86

File: ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java (hunk 1 above)
Comment (mmajerni) [reply ui-tests/src/test/java/io/syndesis/qe/steps/datavirtualization/DataVirtualizationSteps.java]: fixed

File: ui-tests/src/test/resources/features/other/data-virtualization.feature (hunk 4 above)
Comment (mmajerni) [reply ui-tests/src/test/resources/features/other/data-virtualization.feature]: Added one row to contact table and five to todo table. Checking contact table without offset and todo table with offset.

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by mkralik3** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**